lib/AIEFindFlows.cpp

```diff
@@ -124,29 +210,108 @@ class ConnectivityAnalysis {
         // Incoming packets cannot match this rule. Skip it.
         continue;
       }
-      matched = true;
+      // matched = true;
       MaskValue newMaskValue = std::make_pair(
           maskValue.first | nextMaskValue.first,
           maskValue.second | (nextMaskValue.first & nextMaskValue.second));
       auto nextConnection = getConnectionThroughWire(switchOp, nextPort);
 
       // If there is no wire to follow then bail out.
-      if (!nextConnection.hasValue())
+      // then add to antenna list
+      if (!nextConnection.hasValue()) {
+        antennaIndex.push_back(g.get_index(switchOp, nextPort, true, true));
         continue;
+      }
 
+      g.add_child(switchOp, nextPort, nextConnection.getValue().first,
+                  nextConnection.getValue().second, false);
       worklist.push_back(
           std::make_pair(nextConnection.getValue(), newMaskValue));
     }
     return worklist;
   }
 
+  void detect_antenna(Graph &g, std::vector<int> connectedTilesIndexList,
+                      std::vector<int> antennaIndexList) const {
+
+    std::vector<int> path_buffer;
+    // create valid path from connected tiles
+    std::vector<int> valid_path;
```

Comment:
Isn't this just the set of connected tiles?  Would a set-like container be more appropriate?  See: https://llvm.org/docs/ProgrammersManual.html#picking-the-right-data-structure-for-a-task